internal/pkg/manifests/receive/builder.go

```diff
@@ -56,25 +61,40 @@ type IngesterOptions struct {
 	StorageSize    resource.Quantity
 	ObjStoreSecret corev1.SecretKeySelector
 	Retention      string
+	ExternalLabels map[string]string
 }
 
 // RouterOptions for Thanos Receive router
 type RouterOptions struct {
 	manifests.Options
+	ReplicationFactor int32
+	ExternalLabels    map[string]string
 }
 
 // HashringOptions for Thanos Receive hashring
 type HashringOptions struct {
 	manifests.Options
+	// DesiredReplicationFactor is the desired replication factor for the hashrings.
+	DesiredReplicationFactor int32
+	// HashringSettings is the configuration for the hashrings.
+	// The key should be the name of the Service that the hashring is associated with.
 	HashringSettings map[string]HashringMeta
 }
 
 // HashringMeta represents the metadata for a hashring.
 type HashringMeta struct {
-	OriginalName             string
-	Replicas                 int32
-	Tenants                  []string
-	TenantMatcherType        TenantMatcher
+	// OriginalName is the original name of the hashring
+	OriginalName string
+	// DesiredReplicasReplicas is the desired number of replicas for the hashring
+	DesiredReplicasReplicas int32
+	// Tenants is a list of tenants that match on this hashring.
+	Tenants []string
+	// TenantMatcherType is the type of tenant matching to use.
+	TenantMatcherType TenantMatcher
+	// Priority is the priority of the hashring which is used for sorting.
+	// If Priority is the same, the hashring will be sorted by name.
+	Priority int
```

Comment:
Not sure I get the use of this Priority

Reply:
So this isn't exposed to a user (might not be ever), but essentially, we need some way to keep the hashrings in a specific order. A tenant could match multiple hashrings and they will stop at the first hashring they match with, hence the need for some form of priority.

Right now the controller maintains priority by order of slice index so this could be a slice of structs vs a map but I think the priority field gives us a bit more control. wdyt?

Reply:
Yup make sense, we might expose it